video/vdu_stream_processor.h

```diff
@@ -200,27 +208,19 @@ uint32_t VDUStreamProcessor::readIntoBuffer(uint8_t * buffer, uint32_t length, u
 		debug_log("readIntoBuffer: buffer is null\n\r");
 		return remaining;
 	}
-	auto t = xTaskGetTickCountFromISR();
-	auto now = t;
-	auto timeCheck = pdMS_TO_TICKS(timeout);
 
 	while (remaining > 0) {
-		now = xTaskGetTickCountFromISR();
-		if (now - t > timeCheck) {
-			debug_log("readIntoBuffer: timed out\n\r");
-			return remaining;
-		}
-		auto available = inputStream->available();
-		if (available > 0) {
-			if (available > remaining) {
-				available = remaining;
+		auto read = inputStream->readBytes(buffer, remaining);
+		if (read == 0) {
+			// timed out - perform a single retry
+			read = inputStream->readBytes(buffer, remaining);
+			if (read == 0) {
+				debug_log("readIntoBuffer: timed out\n\r");
+				return remaining;
```

Comment:
this function now uses a simpler method of retrying than before.  it makes use of the underlying comms system timeout, using in a single explicit retry in the event of no data being read, as testing showed that we would occasionally see false reports of timeouts

removing the explicit timeout tracking here makes this function simpler and thus slightly more performant